commit 083dd95a0315c18b0f1c7b4e807f67b62a7a7e0f
Author: Bernhard Urban <bernhard.urban@xamarin.com>
Date:   Thu Jul 19 13:12:54 2018 +0200

    [TermInfo] fix reading integer value for new file format (#9520)
    
    16bit vs. 32bit only applies for offsets, not for actual entries in the
    file format.
    
    Fixes https://github.com/mono/mono/issues/8340

diff --git a/mcs/class/corlib/System/TermInfoReader.cs b/mcs/class/corlib/System/TermInfoReader.cs
index 2be4627e791..bb36aca5e4c 100644
--- a/mcs/class/corlib/System/TermInfoReader.cs
+++ b/mcs/class/corlib/System/TermInfoReader.cs
@@ -173,7 +173,7 @@ namespace System {
 				offset++;
 
 			offset += ((int) number) * intOffset;
-			return GetInteger (buffer, offset);
+			return GetInt16 (buffer, offset);
 		}
 
 		public string Get (TermInfoStrings tstr)
@@ -222,27 +222,6 @@ namespace System {
 			return (short) (uno + dos * 256);
 		}
 
-		int GetInt32 (byte [] buffer, int offset)
-		{
-			int b1 = (int) buffer [offset];
-			int b2 = (int) buffer [offset + 1];
-			int b3 = (int) buffer [offset + 2];
-			int b4 = (int) buffer [offset + 3];
-			if (b1 == 255 && b2 == 255 && b3 == 255 && b4 == 255)
-				return -1;
-
-			return b1 + b2 << 8 + b3 << 16 + b4 << 24;
-		}
-
-		int GetInteger (byte [] buffer, int offset)
-		{
-			if (intOffset == 2)
-				return GetInt16 (buffer, offset);
-			else
-				// intOffset == 4
-				return GetInt32 (buffer, offset);
-		}
-
 		string GetString (byte [] buffer, int offset)
 		{
 			int length = 0;
